packages/valory/skills/abstract_round_abci/tests/test_tools/test_rounds.py

```diff
@@ -0,0 +1,689 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+
+"""Test the `rounds` test tool module of the skill."""
+
+
+import re
+from enum import Enum
+from typing import Any, FrozenSet, Generator, List, Optional, Tuple, Type, cast
+from unittest.mock import MagicMock
+
+import pytest
+from hypothesis import given, settings
+from hypothesis import strategies as st
+
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciAppDB,
+    BaseSynchronizedData,
+    BaseTxPayload,
+)
+from packages.valory.skills.abstract_round_abci.test_tools.rounds import (
+    BaseCollectDifferentUntilAllRoundTest,
+    BaseCollectDifferentUntilThresholdRoundTest,
+    BaseCollectSameUntilAllRoundTest,
+    BaseCollectSameUntilThresholdRoundTest,
+    BaseOnlyKeeperSendsRoundTest,
+    BaseRoundTestClass,
+    BaseVotingRoundTest,
+    DummyCollectDifferentUntilAllRound,
+    DummyCollectDifferentUntilThresholdRound,
+    DummyCollectSameUntilAllRound,
+    DummyCollectSameUntilThresholdRound,
+    DummyEvent,
+    DummyOnlyKeeperSendsRound,
+    DummySynchronizedData,
+    DummyTxPayload,
+    DummyVotingRound,
+    MAX_PARTICIPANTS,
+    get_dummy_tx_payloads,
+    get_participants,
+)
+from packages.valory.skills.abstract_round_abci.tests.test_common import last_iteration
+
+
+settings(deadline=5000)
+
+
+def test_get_participants() -> None:
+    """Test `get_participants`."""
+    participants = get_participants()
+    assert isinstance(participants, frozenset)
+    assert all(isinstance(p, str) for p in participants)
+    assert len(participants) == MAX_PARTICIPANTS
+
+
+class DummyTxPayloadMatcher:
+    """A `DummyTxPayload` matcher for assertion comparisons."""
+
+    expected: DummyTxPayload
+
+    def __init__(self, expected: DummyTxPayload) -> None:
+        """Initialize the matcher."""
+        self.expected = expected
+
+    def __repr__(self) -> str:
+        """Needs to be implemented for better assertion messages."""
+        return (
+            "DummyTxPayload("
+            f"id={repr(self.expected.id_)}, "
+            f"round_count={repr(self.expected._round_count)}, "
+            f"sender={repr(self.expected.sender)}, "
+            f"value={repr(self.expected._value)}, "
+            f"vote={repr(self.expected._vote)}"
+            ")"
+        )
+
+    def __eq__(self, other: Any) -> bool:
+        """The method that will be used for the assertion comparisons."""
+        return (
+            self.expected._round_count == other._round_count
+            and self.expected.sender == other.sender
+            and self.expected._value == other._value
+            and self.expected._vote == other._vote
+        )
```

Comment:
first instinct: shouldn't this method be present on the actual payload? Turns out `BaseTxPayload` compares the assigned `uuid4` so for sure no payload should even be considered the same, might as well use `is` instead of `==`. Good practice is to implement a type check and return (not raise) `NotImplemented` if not the same to delegate comparison to `other`.

https://github.com/valory-xyz/open-autonomy/blob/20551f16131c8de0144729f2422bcca91e2210b6/packages/valory/skills/abstract_round_abci/base.py#L263-L272

Do you think vote could also be called `value`? That would simplify matters. I know it's outside the scope of this PR, but added it as one of the notes on the [[WD] refactoring open-autonomy core packages](https://docs.google.com/document/d/1kEC-GctRTU7rOuJ7q18-t13OQQcAwN2QtLhd7-WUlTw). If you have any more ideas / considerations, please add them as well

Reply:
exactly why it's added to the WD